### PR TITLE
[Communication] Cleanup the Recording Test Infrastructure for Identity, Chat and Phone Numbers

### PR DESCRIPTION
**Pull request by RezaJooyandeh**
# What is in this change?
* Removing sanitization for things that do not need to be sanitized.
* Removing duplicated sanitization code.
* Extracting the shared sanitization code to Common.Test and ensure resource names are sanitized across all tests.
* Remove redundant environment variables. AccessKey and Endpoint could be extracted from ConnectionString.

### CODE REVIEW

File: sdk/communication/Azure.Communication.Chat/tests/SessionRecords/ChatClientsLiveTests/ReadReceiptGSAsyncAsync.json [hunk 1]
```diff
@@ -129,75 +141,81 @@
...
+        "expiresOn": "2021-02-13T03:08:47.9804957\u002B00:00"
       }
     },
     {
-      "RequestUri": "https://chat-sdktester-e2e.communication.azure.com/chat/threads?api-version=2020-11-01-preview3",
+      "RequestUri": "https://sanitized.communication.azure.com/chat/threads?api-version=2020-11-01-preview3",
       "RequestMethod": "POST",
       "RequestHeaders": {
         "Accept": "application/json",
         "Authorization": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c",
```
Comment (turalf) [inline sdk/communication/Azure.Communication.Chat/tests/SessionRecords/ChatClientsLiveTests/ReadReceiptGSAsyncAsync.json:162]: Is this intentional that Authorization header is not sanitized?

File: sdk/communication/Azure.Communication.Chat/tests/SessionRecords/ChatClientsLiveTests/ReadReceiptGSAsyncAsync.json [hunk 2]
```diff
@@ -240,82 +261,91 @@
       "ResponseHeaders": {
         "api-supported-versions": "2020-09-21-preview2, 2020-11-01-preview3, 2021-01-27-preview4",
         "Content-Type": "application/json; charset=utf-8",
-        "Date": "Fri, 05 Feb 2021 04:50:19 GMT",
-        "Location": "https://chat-sdktester-e2e.communication.azure.com/chat/threads/19%3AMGFVz7lbRHfSeRIAyD9rXZqx5i4TAGdAemi1DcEuvRE1@thread.v2/messages/1612500620132",
-        "MS-CV": "w5Y7nw8l0kGtCExOtkDFsg.0",
+        "Date": "Fri, 12 Feb 2021 03:08:49 GMT",
+        "Location": "https://smstestapp.communication.azure.com/chat/threads/19%3AKKlEhjwvup9_n9n1Qlq5itO5y-ZitOvWWcX11iMBlKA1@thread.v2/messages/1613099329845",
```
Comment (turalf) [inline sdk/communication/Azure.Communication.Chat/tests/SessionRecords/ChatClientsLiveTests/ReadReceiptGSAsyncAsync.json:265]: I think resource names are fine to be left unsansitized as they get destroyed, but for consistency sake should this be sanitized too, or unsanitize the rest?

File: sdk/communication/Azure.Communication.Chat/tests/SessionRecords/ChatClientsLiveTests/ReadReceiptGSAsyncAsync.json (hunk 1 above)
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Chat/tests/SessionRecords/ChatClientsLiveTests/ReadReceiptGSAsyncAsync.json:162]: nope that is a mistake

File: sdk/communication/Azure.Communication.Chat/tests/SessionRecords/ChatClientsLiveTests/ReadReceiptGSAsyncAsync.json (hunk 2 above)
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Chat/tests/SessionRecords/ChatClientsLiveTests/ReadReceiptGSAsyncAsync.json:265]: Yeah the Location needs sanitization. Resource name I think should be sanitized

Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Chat/tests/SessionRecords/ChatClientsLiveTests/ReadReceiptGSAsyncAsync.json]: I pushed a new update that sanitize the location header as well

Comment (juancamilor) [reply sdk/communication/Azure.Communication.Chat/tests/SessionRecords/ChatClientsLiveTests/ReadReceiptGSAsyncAsync.json]: FYI @RezaJooyandeh they are not required to be sanitized, along with IP adress, or IDs, we ran that few times by the azure folks long ago.

Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Chat/tests/SessionRecords/ChatClientsLiveTests/ReadReceiptGSAsyncAsync.json]: You are right, looking around in other libraries as well it seems that we do not sanitize the urls anymore. I will remove it

File: sdk/communication/Azure.Communication.PhoneNumbers/tests/Infrastructure/PhoneNumbersClientRecordedTestSanitizer.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        private static readonly Regex _phoneNumberRegEx = new Regex(@"[\\+]?[0-9]{11,15}", RegexOptions.Compiled);
+        private static readonly Regex _guidRegEx = new Regex(@"(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}", RegexOptions.Compiled);
+
+        public PhoneNumbersClientRecordedTestSanitizer()
+        {
+            JsonPathSanitizers.Add("$..phonePlanId");
+            JsonPathSanitizers.Add("$..phonePlanGroupId");
+            JsonPathSanitizers.Add("$..phonePlanIds[:]");
+        }
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/tests/Infrastructure/PhoneNumbersClientRecordedTestSanitizer.cs:18]: Once we migrate to the new service, we should remove guid sanitization and id sanitization should be removed too. Only phone numbers need to be replaced with a fake number

File: sdk/communication/Azure.Communication.PhoneNumbers/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs [hunk 4]
```diff
@@ -247,7 +249,8 @@ public async Task CreateReservationErrorState(string? locale)
         [Test]
         [TestCase(null)]
         [TestCase("en-US")]
-        public async Task CreateReservation(string? locale)
+        [Ignore("TollFreeNumbersAreDisabledTemporarily")]
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs:252]: We are going to replace these anyways. I disabled them as they are disabled for now and are breaking the tests

File: sdk/communication/Azure.Communication.Identity/tests/SessionRecords/CommunicationIdentityClientLiveTests/GettingTokenWithMultipleScopesWithConnectionString.json [hunk 5]
```diff
@@ -23,32 +23,33 @@
...
       "ResponseBody": {
         "identity": {
-          "id": "Sanitized"
+          "id": "8:acs:cb58c3bb-a8ef-4e54-8c12-4a8e652eba58_00000008-57be-0fd2-6a0b-343a0d005880"
         }
       }
     },
     {
-      "RequestUri": "https://sanitized.communication.azure.com/identities/sanitized/:issueAccessToken?api-version=2021-03-07",
+      "RequestUri": "https://chitchat-test.communication.azure.com/identities/8%3Aacs%3Acb58c3bb-a8ef-4e54-8c12-4a8e652eba58_00000008-57be-0fd2-6a0b-343a0d005880/:issueAccessToken?api-version=2021-03-07",
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Identity/tests/SessionRecords/CommunicationIdentityClientLiveTests/GettingTokenWithMultipleScopesWithConnectionString.json:42]: Is there a reason why we used to sanitize user ids in some places but not in others?

File: sdk/communication/Azure.Communication.Sms/tests/SmsClientRecordedTestSanitizer.cs [hunk 6]
```diff
@@ -7,13 +7,10 @@ namespace Azure.Communication.Sms.Tests
...
     public class SmsClientRecordedTestSanitizer : CommunicationRecordedTestSanitizer
     {
-        public SmsClientRecordedTestSanitizer() : base()
+        public SmsClientRecordedTestSanitizer()
         {
-            JsonPathSanitizers.Add("$..token");
-            JsonPathSanitizers.Add("$..id");
             JsonPathSanitizers.Add("$..from");
             JsonPathSanitizers.Add("$..to");
-            JsonPathSanitizers.Add("$..messageId");
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Sms/tests/SmsClientRecordedTestSanitizer.cs:16]: Guess we didn't really needed to sanitize what you just removed?

Should we re-record the Sms tests too?

**Comment by DominikMe** [comment]
@RezaJooyandeh 
This is the failure in the CI

```
Uri doesn't match:
    request <https://sanitized.communication.azure.com/administration/phonenumbers/searches/a83f9c32-d87d-4e82-ac33-409fa0803bce?api-version=2020-07-20-preview1>
    record  <https://sanitized.communication.azure.com/administration/phonenumbers/searches/Sanitized?api-version=2020-07-20-preview1>
```

File: sdk/communication/Azure.Communication.Sms/tests/SmsClientRecordedTestSanitizer.cs (hunk 6 above)
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Sms/tests/SmsClientRecordedTestSanitizer.cs:16]: Right now, we cannot record everything. There are some server issues

File: sdk/communication/Azure.Communication.Identity/tests/SessionRecords/CommunicationIdentityClientLiveTests/GettingTokenWithMultipleScopesWithConnectionString.json (hunk 5 above)
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Identity/tests/SessionRecords/CommunicationIdentityClientLiveTests/GettingTokenWithMultipleScopesWithConnectionString.json:42]: use ids are not to be sanitized at all. Where did you see it being sanitized?

**Comment by RezaJooyandeh** [comment]
@DominikMe  Some of tests that were in playback mode only needed to be updated. We cannot record them now due to the disablement. So I manually updated the recordings

File: sdk/communication/Azure.Communication.PhoneNumbers/tests/samples/Sample2_PhoneNumberAdministrationClient.cs [hunk 7]
```diff
@@ -467,6 +467,7 @@ public async Task GeographicalAreaCodesAsync()
 
         [Test]
         [SyncOnly]
+        [PlaybackOnly("TollFreeNumbersAreDisabledTemporarily")]
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.PhoneNumbers/tests/samples/Sample2_PhoneNumberAdministrationClient.cs:470]: Toll Free numbers are disabled, but also we are going to replace these tests completely for the new API anyways, so it is fine to change them to playback.

File: sdk/communication/Azure.Communication.Sms/tests/SmsClientRecordedTestSanitizer.cs (hunk 6 above)
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Sms/tests/SmsClientRecordedTestSanitizer.cs:16]: We can rerecord them later. In fact @ankitarorabit has to do them with all the new changes in the swagger anyways

**Review by DominikMe — APPROVED** [review]

**Pull request merged**